api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java

```diff
@@ -43,4 +45,18 @@ public void shouldFilterAgainstFormFields() {
 		assertEquals(0, (Object)dao.getForms(null, false, Collections.emptyList(), null, formFields, formFields, Arrays.asList(new Field(3))).size());
 		
 	}
+
+	@Test
+	public void shouldGetFormFieldsByForm() {
+		Form form = new Form(2); 
+		List<FormField> formFields = dao.getFormFields(form);
+
+		assertNotNull(formFields);
+
+		int expectedSize = 2;
```

Comment:
Shouldn't the above be final?

Reply:
Updated to make it final. I would rarely make local variables final, do you prefer to do this by default for all variables that are intended to be immutable or are there specific scenarios you want them to be final?

Reply:
Is there any reason why you rarely make local variables final?

Reply:
I guess the main reason is other teams I have been on have only used them at class and field level, not so much method and local levels. My preference has been to omit it is that I find it more verbose / less readable with finals. I can see the reasons behind doing it though as it prevents it being changed unintentionally. Regardless, happy to fall in line with the openmrs standard practice